apps/remote_control/lib/lexical/remote_control/build/error.ex

```diff
@@ -12,19 +12,83 @@ defmodule Lexical.RemoteControl.Build.Error do
     detail_diagnostics = detail_diagnostics(detail)
     error = message_info_to_binary(message_info, token)
     error_diagnostics = parse_error_to_diagnostics(context, error, token)
-    error_diagnostics ++ detail_diagnostics
+    uniq(error_diagnostics ++ detail_diagnostics)
   end
 
   def parse_error_to_diagnostics(context, message_info, token) do
-    diagnostic = %Diagnostic{
-      file: nil,
-      severity: :error,
-      position: context_to_position(context),
-      compiler_name: "Elixir",
-      message: message_info_to_binary(message_info, token)
-    }
+    parse_error_diagnostic_functions = [
+      &build_end_line_diagnostics/3,
+      &build_start_line_diagnostics/3,
+      &build_hint_diagnostics/3
+    ]
 
-    [diagnostic]
+    Enum.flat_map(parse_error_diagnostic_functions, & &1.(context, message_info, token))
+  end
+
+  defp uniq(diagnostics) do
+    # We need to uniq by position because the same position can be reported
+    # and the `end_line_diagnostic` is always the precise one
+    position_key = fn
+      %Diagnostic{position: position} when is_tuple(position) ->
```

Comment:
Technically, this means you will eliminate multiple issues on the same line. Not sure if the compiler can handle that, but calling it out nonetheless.

Reply:
Also, why not just match on {line, _column}?

Reply:
> Technically, this means you will eliminate multiple issues on the same line. Not sure if the compiler can handle that, but calling it out nonetheless.


Yes, but what I eliminated was the information generated by ourselves. In fact, there is only one piece of original information provided by the compiler.